src/mcp_server/tools/observability_vllm_tools.py

```diff
@@ -473,3 +473,186 @@ def calculate_metrics(
+
+        gpu_info: Dict[str, Any] = {
+            "total_gpus": 0,
+            "vendors": [],
+            "models": [],
+            "temperatures": [],
+            "power_usage": [],
+        }
+
+        # Attempt to infer GPU count and temperatures from DCGM metrics
+        try:
+            response = requests.get(
+                f"{PROMETHEUS_URL}/api/v1/query",
+                headers=headers,
+                params={"query": "DCGM_FI_DEV_GPU_TEMP"},
+                verify=VERIFY_SSL,
+                timeout=30,
+            )
+            if response.status_code == 200:
+                result = response.json().get("data", {}).get("result", [])
+                gpu_info["total_gpus"] = len(result)
+                temperatures = [float(series["value"][1]) for series in result if series.get("value")]
+                gpu_info["temperatures"] = temperatures
+                if temperatures:
+                    gpu_info["vendors"] = ["NVIDIA"]
+                    gpu_info["models"] = ["GPU"]
+        except Exception as e:
+            # Log but continue with defaults
+            logger.warning(f"Error querying GPU metrics: {e}")
+
+        return _resp(json.dumps(gpu_info))
+    except Exception as e:
+        error = MCPException(
+            message=f"Failed to get GPU info: {str(e)}",
+            error_code=MCPErrorCode.PROMETHEUS_ERROR,
+            recovery_suggestion="Verify Prometheus/Thanos connectivity and DCGM exporter availability."
+        )
+        return error.to_mcp_response()
+
+
+def get_deployment_info(namespace: str, model: str) -> List[Dict[str, Any]]:
+    """Get deployment info for a model in a namespace (equivalent to /deployment-info).
+
+    Heuristic:
+    - Query kube_pod_info in namespace to ensure activity
+    - Look at vLLM cache config metric timeline over the last week to infer start time
+    """
+    try:
+        validate_required_params(namespace=namespace, model=model)
+    except ValidationError as e:
+        return e.to_mcp_response()
+
+    try:
+        headers = {"Authorization": f"Bearer {THANOS_TOKEN}"} if THANOS_TOKEN else {}
+
+        # Initial probe: any pods in namespace
+        try:
+            query = f'kube_pod_info{{namespace="{namespace}"}}'
+            response = requests.get(
+                f"{PROMETHEUS_URL}/api/v1/query",
+                headers=headers,
+                params={"query": query},
+                verify=VERIFY_SSL,
+                timeout=30,
+            )
+            response.raise_for_status()
+            result = response.json().get("data", {}).get("result", [])
+        except Exception as e:
+            logger.warning(f"Error probing namespace pods: {e}")
+            result = []
+
+        current_time = datetime.utcnow()
+        is_new_deployment = False
+        deployment_date: Optional[str] = None
+
+        if result:
+            # Check vLLM cache config metric timeline for last 7 days
```

Comment:
can we move this logic into core ? and mcp tool just call them

Reply:
Done